Vagrantfile

```diff
@@ -7,18 +62,76 @@ Vagrant.configure("2") do |config|
     end
     v.memory = 4096
   end
- [
-   %w{centos6.5 chef/centos-6.5},
-   %w{centos7   chef/centos-7.0},
-   %w{ubuntu14  ubuntu/trusty64},
-   %w{ubuntu12  ubuntu/precise64},
-   %w{freebsd10 chef/freebsd-10.0}
- ].each do |machine|
-  (name, box) = machine
+
+  config.vm.provider :aws do |aws, override|
+
+    # Required. Credentials for aws api. vagrant-aws will error if these are unset.
+    aws.access_key_id = ENV['AWS_ACCESS_KEY_ID']
+    aws.secret_access_key = ENV['AWS_SECRET_ACCESS_KEY']
+
+    # Required. Name of aws keypair for launching and accessing the ec2 instance.
+    if [ ENV['AWS_KEYPAIR_NAME'] ]
```

Comment:
nice catch. for stock amis, the keypair is required or you end up with a vm you cant login to. the vagrant-aws plugin issues a warning when the keypair is not set but does not fail hard. its possible that someone could build an ami that has preset creds that obviate the need for the keypair. given that a clear warning is issued by vagrant-aws, i removed the "required" label and kept the if block. let me know if there's a better route.